kernel/tests/golden_tables.rs

```diff
@@ -153,37 +153,35 @@ async fn latest_snapshot_test(
     table: Table,
     expected_path: Option<PathBuf>,
 ) -> Result<(), Box<dyn std::error::Error>> {
-    let snapshot = table.snapshot(&engine, None).unwrap();
+    let snapshot = table.snapshot(&engine, None)?;
 
-    let scan = snapshot.into_scan_builder().build().unwrap();
-    let scan_res = scan.execute(&engine).unwrap();
+    let scan = snapshot.into_scan_builder().build()?;
+    let scan_res = scan.execute(&engine)?;
     let batches: Vec<RecordBatch> = scan_res
-        .into_iter()
-        .map(|sr| {
-            let data = sr.raw_data.unwrap();
-            let record_batch = to_arrow(data).unwrap();
+        .map(|sr| -> DeltaResult<_> {
+            let sr = sr?;
```

Comment:
Would `map_ok` simplify things, by not requiring `Ok` at L171 and L173 below?
Actually, since we're materializing, `process_result` might be even better?

Reply:
For a similar reason to my comment [here](https://github.com/delta-incubator/delta-kernel-rs/pull/340/files/7ce4593a9f679387489c5b4bf9d9c2aa8eab8e9b#r1781996948), the Result input and Result output leads to `process_results` becoming `DeltaResult<DeltaResult<_>>`, requiring a `??` at the end. It also felt harder to reason about.